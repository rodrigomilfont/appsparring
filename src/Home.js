// @flow

import React from 'react';
// import logo from './logo.svg';

const Home = () => (
  <div className="Home">
    <header className="App-header">
      <h1 className="App-title">Welcome to React</h1>
      {/* <img src={logo} className="App-logo" alt="logo" /> */}
    </header>
    <h3>Welcome h3</h3>
    <h2>Welcome h2</h2>
    <h4>
      {' '}
      <a>Blah</a> {' '}
    </h4>
    <h5>H5</h5>
    <p className="App-intro">
      To get started, edit <code>src/.js</code> and save to reload.{' '}
    </p>{' '}
  </div>
);

export default Home;
