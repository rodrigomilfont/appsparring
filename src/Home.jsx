// @flow

import React from 'react';
import Header from './components/header/Header';

const Home = (props) => (
  <div className="grid">
    <Header {...props } />
    <aside className="sidebar-left">
      <article>
        <h4>
          <a href="#deded">Blah</a>
        </h4>
        <h2>Welcome h2</h2>
        <h5>Welcome H5</h5>
      </article>
    </aside>
    <main className="content">
      <article>
        <h3>Welcome h3</h3>
        <p>
          Is a non-negative <strong>percentage</strong> value, relative to the
          block size of the grid container. If the size of the grid container
          depends on the size of its tracks, then the percentage must be treated
          as auto. The intrinsic size contributions of the track may be adjusted
          to the size of the grid container, and increase the final size of the
          track by the minimum amount that would result in honoring the
          percentage.
        </p>{' '}
        <p>
          Is a non-negative <strong>percentage</strong> value, relative to the
          block size of the grid container. If the size of the grid container
          depends on the size of its tracks, then the percentage must be treated
          as auto. The intrinsic size contributions of the track may be adjusted
          to the size of the grid container, and increase the final size of the
          track by the minimum amount that would result in honoring the
          percentage.
        </p>{' '}
        <p>
          Is a non-negative <strong>percentage</strong> value, relative to the
          block size of the grid container. If the size of the grid container
          depends on the size of its tracks, then the percentage must be treated
          as auto. The intrinsic size contributions of the track may be adjusted
          to the size of the grid container, and increase the final size of the
          track by the minimum amount that would result in honoring the
          percentage.
        </p>
      </article>
    </main>
    <aside className="sidebar-rigth">
      <p>
        <span className="bt_nav">
          <a className="bt_home" href="#home">
            Home
          </a>
        </span>
      </p>
    </aside>
    <footer className="footer">
      <p className="footer-text">
        To get started, edit <code>src/.js</code> and save to reload.
      </p>
    </footer>
  </div>
);

export default Home;
