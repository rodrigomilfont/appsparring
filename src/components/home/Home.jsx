import React from 'react';
import { Link } from 'react-router-dom';
import Header from '../header/Header';
import donaGata from '../header/donaGata84.jpg';
import Felix from '../header/felix_the_cat.svg';

const Home = props => (
  <div className="grid">
    <Header {...props} />
    <aside className="sidebar-left">
      <article>
        <ul>
          <li>
            <Link to="/items">Miercado Libre</Link>
          </li>
          <li>
            <Link to="/items">Miercado Libre</Link>
          </li>
        </ul>
      </article>
    </aside>
    <main className="content">
      <img src={donaGata} alt="" />
      <article>
        <h1>Home</h1>
      </article>
      <img src={Felix} alt="Felix" width="200" />
    </main>
    <aside className="sidebar-rigth">
      <p>
        <span className="bt_nav">
          <Link to="/">Home Items</Link>
        </span>
      </p>
    </aside>
    <footer className="footer">
      <p className="footer-text">Footer</p>
    </footer>
  </div>
);

export default Home;
