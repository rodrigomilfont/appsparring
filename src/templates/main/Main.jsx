import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const Main = props => <div className="main">{props.children}</div>;

Main.propTypes = {
  children: PropTypes.element,
};

Main.defaultProps = {
  children: function noop() {},
};

export default Main;
