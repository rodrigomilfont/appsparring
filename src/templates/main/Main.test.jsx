import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import Main from './Main';
import Home from '../../components/home/Home';
import ComposeLayoutRoute from '../composeLayoutRoute';

Enzyme.configure({ adapter: new Adapter() });

it('renders Main', () => {
  const wrapper = shallow(
    <ComposeLayoutRoute exact path="/" layout={Main} component={Home} />,
  );
  expect(wrapper).toMatchSnapshot();
});

it('render MainLayout', () => {
  const wrapper = shallow(
    <Main>
      <div className="class-intern-main" />
    </Main>,
  );
  expect(wrapper.find('.class-intern-main').length).toEqual(1);
  // console.log(wrapper.debug());
  expect(wrapper).toMatchSnapshot();
});
