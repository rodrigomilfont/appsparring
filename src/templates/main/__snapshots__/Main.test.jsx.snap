// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MainLayout 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main>
    <div
      className="class-intern-main"
    />
  </Main>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="class-intern-main"
      />,
      "className": "main",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "class-intern-main",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="class-intern-main"
        />,
        "className": "main",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "class-intern-main",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders Main 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComposeLayoutRoute
    component={[Function]}
    exact={true}
    layout={[Function]}
    path="/"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "exact": true,
      "path": "/",
      "render": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "exact": true,
        "path": "/",
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
